src/Microsoft.AspNetCore.Mvc.Core/ProblemDetails.cs

```diff
@@ -17,28 +19,47 @@ public class ProblemDetails
         /// (e.g., using HTML [W3C.REC-html5-20141028]).  When this member is not present, its value is assumed to be
         /// "about:blank".
         /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public string Type { get; set; }
 
         /// <summary>
         /// A short, human-readable summary of the problem type.It SHOULD NOT change from occurrence to occurrence
         /// of the problem, except for purposes of localization(e.g., using proactive content negotiation;
         /// see[RFC7231], Section 3.4).
         /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public string Title { get; set; }
 
         /// <summary>
         /// The HTTP status code([RFC7231], Section 6) generated by the origin server for this occurrence of the problem.
         /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public int? Status { get; set; }
 
         /// <summary>
         /// A human-readable explanation specific to this occurrence of the problem.
         /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public string Detail { get; set; }
 
         /// <summary>
         /// A URI reference that identifies the specific occurrence of the problem.It may or may not yield further information if dereferenced.
         /// </summary>
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
         public string Instance { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="IDictionary{TKey, TValue}"/> for extension members.
+        /// <para>
+        /// Problem type definitions MAY extend the problem details object with additional members. Extension members appear in the same namespace as
+        /// other members of a problem type.
+        /// </para>
+        /// </summary>
+        /// <remarks>
+        /// The round-tripping behavior for <see cref="Extension"/> is determined by the implementation of the Input \ Output formatters.
+        /// In particular, complex types or collection types may not round-trip in the original format when using the built-in JSON or XML formatters.
+        /// </remarks>
+        [JsonExtensionData]
+        public IDictionary<string, object> Extension { get; } = new Dictionary<string, object>(StringComparer.Ordinal);
```

Comment:
Since this property name doesn't show up in the serialization and we give dictionaries a plural name, could this be `Extensions`?